flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/deployment/SessionReconciler.java

```diff
@@ -133,7 +135,16 @@ protected void rollback(FlinkDeployment deployment, Context ctx, Configuration o
         FlinkDeploymentSpec rollbackSpec = reconciliationStatus.deserializeLastStableSpec();
         Configuration rollbackConfig =
                 configManager.getDeployConfig(deployment.getMetadata(), rollbackSpec);
-        upgradeSessionCluster(deployment, rollbackSpec, rollbackConfig);
+
+        deleteSessionCluster(deployment, observeConfig);
+        deploy(
```

Comment:
Why in the rollback we do not have to record the target upgrade status ? I think the rollback should deploy the stable spec with the corresponding generation, WDYT

Reply:
I also found that the general flow of the deploy become the 
- record the target upgrade/firstDeploy in status
- deploy
- update the deployed spec

So I think we could include it in the `deploy` directly

Reply:
The rollback logic doesn't assume anything about the generation etc so there is not really a need to record it.

Also we cannot put this in `deploy` directly because deploy is used for spec update, JM recovery, and rollback. And the target generation will only be useful for spec change.